src/System.Private.CoreLib/shared/System/IO/Path.cs

```diff
@@ -812,8 +812,12 @@ public static string GetRelativePath(string relativeTo, string path)
 
         private static string GetRelativePath(string relativeTo, string path, StringComparison comparisonType)
         {
-            if (string.IsNullOrEmpty(relativeTo)) throw new ArgumentNullException(nameof(relativeTo));
-            if (PathInternal.IsEffectivelyEmpty(path.AsSpan())) throw new ArgumentNullException(nameof(path));
+            if (relativeTo == null)
+                throw new ArgumentNullException(nameof(relativeTo));
+            if (relativeTo.Length == 0)
+                throw new ArgumentException(nameof(relativeTo));
+            if (PathInternal.IsEffectivelyEmpty(path.AsSpan()))
+                throw new ArgumentNullException(nameof(path));
```

Comment:
This has the same issue as the error checking for `relativeTo` argument: It will throw `ArgumentNullException` for non-null empty string, or even for non-empty strings of all spaces (on Windows).